scripts/release-npm.sh

```diff
@@ -19,32 +19,23 @@ for i in "$@"; do
     esac
 done
 
-if [ -z "$TAG" ]; then
-    usage
-    exit 1
+if ! which goreleaser >/dev/null ; then
```

Comment:
The other step in CircleCI that runs GoReleaser cleans up the `dist/` folder afterwards. And it would generate more files than we'd need for NPM (for example, deb executables). So adding the `goreleaser build --snapshot` here because we need to and because this way we can run `./release-npm.sh` on its own, without depending on a previous step

Reply:
Hmm, I see what you mean, and this is why we re-run goreleaser below? This might be a little more complicated than is strictly necessary. I'm not super-familiar with github actions, but is it possible to preserve the dist dir across multiple steps, so that we can simply copy the built-once binaries into an npm package here, making this step a lot shorter and simpler?

Reply:
Yeah there's definitely 2 sides to the trade-off. I don't have a strong opinion which way we go, so I'm happy to approve this in principle. It's certainly a step in the right direction.

I might be doing something wrong, but does this actually set the version tag correctly?

```
./scripts/release-npm.sh --tag=v1.2.3
...

❯ ./dist/snyk-iac-rules_darwin_amd64_v1/snyk-iac-rules --version
snyk-iac-rules version 1.5.4-next
```

Reply:
OK, sounds good, let's go with this method. I'll approve this PR.

Good spot about the v1. It should indeed be present in the pipeline now that we've bumped go to 1.18. I'm assuming you're not using go 1.18, which is why you're not seeing it?

Reply:
I was using go1.18, but an older version of goreleaser - `1.9.2` is the right one and now I get the `_v1` appended